### ISSUE TITLE
[BUG] Space station dimensions seem to break after upgrading E2L server

### ISSUE THREAD

**Issue by Nomad7** [zmaster587/AdvancedRocketry#1756]
[Please fill out the form below and delete the sections in square brackets after reading them]

## Version of Advanced Rocketry
>AdvancedRocketry-1.12.2-1.6.0-190-universal.jar

## Have you verified this is an issue in the latest unstable build
- [x] N/A

> I did try to reproduce this behavior by manually updating our E2L modpack to AR build 193 and experienced what appeared to be the same issue (which has also been present on older versions), but I'm not sure if manual, piecemeal upgrade scenarios are valid.

## Version of LibVulpes
>LibVulpes-1.12.2-0.4.2-69-universal.jar

## Version of Minecraft
>1.12.2

## Does this occur without other mods installed
- [x] N/A
> We're running the Enigmatica 2 Light modpack on our server - unfortunately I don't really have a good way to test AR in isolation in this server/client scenario.

## Crash report or log or visualVM (if applicable)
> Here's the relevant snippet from my client log trimmed to a size pastebin allows: https://pastebin.com/cFTfz7nt
> Here's my whole client log file:
[latest.log](https://github.com/zmaster587/AdvancedRocketry/files/4032926/latest.log)
(I think errors would keep piling up if I didn't cancel my connection attempt)

## Description of the problem
>Space station dimensions appear to break when upgrading mods on the server.
For example: 
1. player logs off while in space station
2. server shuts down
3. mods are updated (including dumping old config/mods/schematics/scripts folders, with manual preservation of any config changes)
4. server is restarted
5. player tries to join server
In this case that player is stuck at the 'logging in' step as errors continue to pile up in the client log (attached).

The relevant item here appears to be:
> java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Could not get provider type for dimension -2, does not exist
But this dimension does indeed exist, and was working fine prior to updating the server mods.

I wouldn't be surprised if this may be related to something else in the E2L pack, and if some manual preservation/repair of dimension provider data is needed to work around this issue I'm happy to do that, but I'm not sure where to look for that sort of thing...

**Comment by zmaster587** [comment]
I don't see any indication dim-2 is being registered, can you send both client and server AR configs?

**Comment by Nomad7** [comment]
Sure thing, here they are (renamed to indicate source and switched to .txt so github doesn't complain):
[advancedRocketry.cfg_client.txt](https://github.com/zmaster587/AdvancedRocketry/files/4032991/advancedRocketry.cfg_client.txt)
[advancedRocketry.cfg_server.txt](https://github.com/zmaster587/AdvancedRocketry/files/4032992/advancedRocketry.cfg_server.txt)

**Comment by zmaster587** [comment]
I noticed that `spaceStationId` differs between the two, before build 192, that would probably be an issue.  Config sync was revamped in 192.  What happens if you change `spaceStationId` from -2 to 99 on the client?

Now that you mention it, I think the space dimension is usually registered before the config is synced, so the space station id may sync, but this problem could still occur.

**Comment by Nomad7** [comment]
Looks like if I set the client config to spaceStationId=-2 (to match the server) I'm able to connect successfully!
(In other news, I left the game trying to join the server while I ate dinner and came back to a >700mb log file :p)

Sounds like the workaround for my current situation is to move our server's space station dimension to 99 to match the client defaults - if that's as easy as renaming the dimension folder and updating the config that's no problem.

I also ran the following test:
- Update AR version on server from 190 to 193 (just replace the .jar in the mods folder)
- Update the AR version on the client from 190 to 193 (same as above)
- With server and client config spaceStationId values set to -2, client was able to connect successfully
- After changing client spaceStationId back to 99 (different than the server) and relaunching the client, got stuck on the 'Logging in' step again

Just an anecdote but this seems like the issue is also present in 193. If there's any additional testing I can do just let me know!
